source/service.go

```diff
@@ -307,7 +307,22 @@ func (sc *serviceSource) extractHeadlessEndpoints(svc *v1.Service, hostname stri
 
 			for _, headlessDomain := range headlessDomains {
 				var ep string
-				if sc.publishHostIP {
+				if sc.publishHostIP && externalIP {
+					node, err := sc.nodeInformer.Lister().Get(pod.Spec.NodeName)
```

Comment:
We can move this query down inside the consequent block at line 306. There's no reason to look up the hosting node if the pod isn't running yet.

Reply:
Yes, sorry about that. That was a typo of mine.